### PR TITLE
Bring ConvModule from mmcv.cnn.bricks

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

`mmcv.cnn.bricks.ConvModule` is used by the `litehrnet` in the 
segmentation and `efficientnet-b0` in classification.
Import that module to work without MMCV.
@kprokofi Please see this when decoupling `LiteHRNet`.

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-US style='font-family:Calibri;font-size:11.0pt'>

<div style='direction:ltr'>

task | dataset | model | ConvModule | test/accuracy
-- | -- | -- | -- | --
MULTI_CLASS_CLS | Medium | Efficientnet-B0 | mmcv.cnn.bricks.ConvModule | 0.7944
  |   |   | otx.algo.modules.conv_module.ConvModule | 0.7944

</div>

</body>

</html>

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/algo/classification/efficientnet_b0.py
Comment (harimkang) [inline src/otx/algo/classification/efficientnet_b0.py:1]: Fix https://github.com/openvinotoolkit/training_extensions/issues/3310

File: src/otx/algo/modules/conv_module.py [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+from .activation import build_activation_layer
+from .conv import build_conv_layer
+from .norm import build_norm_layer
+from .padding import build_padding_layer
+
+if TYPE_CHECKING:
+    from torch.nn.modules.conv import _ConvNd as ConvNd
+
+
+def efficient_conv_bn_eval_forward(bn: BatchNorm, conv: ConvNd, x: torch.Tensor) -> torch.Tensor:
```
Comment (kprokofi) [inline src/otx/algo/modules/conv_module.py:30]: Is it used by our algorithms? Because it is False by default in ConvModule constructor and for LiteHRNet this parameter isn't changed.

File: src/otx/algo/modules/conv_module.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+        """
+        # efficient_conv_bn_eval works for conv + bn
+        # with `track_running_stats` option
+        if efficient_conv_bn_eval and self.norm and isinstance(self.norm, BatchNorm) and self.norm.track_running_stats:
+            self.efficient_conv_bn_eval_forward = efficient_conv_bn_eval_forward
+        else:
+            self.efficient_conv_bn_eval_forward = None
+
+    @staticmethod
+    def create_from_conv_bn(
```
Comment (kprokofi) [inline src/otx/algo/modules/conv_module.py:317]: I also didn't see where it is used

**Review by eugene123tw — APPROVED** [review]

File: src/otx/algo/modules/padding.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+from torch import nn
+
+PADDING_DICT = {
+    "zero": nn.ZeroPad2d,
+    "reflect": nn.ReflectionPad2d,
+    "replicate": nn.ReplicationPad2d,
+}
+
+
+def build_padding_layer(cfg: dict, *args, **kwargs) -> nn.Module:
```
Comment (kprokofi) [inline src/otx/algo/modules/padding.py:18]: standart nn.conv modules have parameter like padding_type. We can just change it when needed. Also, we don't have usecases now, when we need to pad images in convolutions different from zeros. I propose to delete this (remove functionality of constructing padding and use standard settings)

File: src/otx/algo/classification/efficientnet_b0.py
Comment (vinnamkim) [reply src/otx/algo/classification/efficientnet_b0.py:1]: Please wait https://github.com/openvinotoolkit/training_extensions/pull/3370

File: src/otx/algo/modules/conv_module.py (hunk 1 above)
Comment (sungchul1) [reply src/otx/algo/modules/conv_module.py:30]: I think it is for efficientbet-b0, right?

**Review by sungchul1 — APPROVED** [review]

File: src/otx/algo/modules/conv_module.py (hunk 1 above)
Comment (harimkang) [reply src/otx/algo/modules/conv_module.py:30]: Since this is a module that is used across the board regardless of task, I just imported the code from that module file as-is. I'll check it out later and remove it if it's unnecessary.

File: src/otx/algo/modules/conv_module.py (hunk 2 above)
Comment (harimkang) [reply src/otx/algo/modules/conv_module.py:317]: I'll check it out later and remove it if it's unnecessary.

File: src/otx/algo/modules/padding.py (hunk 3 above)
Comment (harimkang) [reply src/otx/algo/modules/padding.py:18]: Yes, I agree. However, I will create a separate PR to remove the unnecessary functions.
